### DISCUSSION TITLE
Question about upcoming v3 Release

### DISCUSSION THREAD

**Discussion by roman-hatz-srf** [maplibre/maplibre-gl-js#1850] [General]
I saw some interesting features in the v3 pre-releases. 
- The version jump means there will be breaking changes? 
- We don't use MapLibre on production yet, does it make sense to work with v3 now? 
- Is there a roadmap when v3 is going to be released roughly? 
Thanks.

**Comment by HarelM** [comment]
You can start using v3 I believe (I'm using it in my product, which is in production and servers 50K users roughly).
The reason we are not releasing it right now is that we want to see if there are more breaking changes we can introduce as part of the v3 release instead of creating a new breaking changes version soon.
We marked the breaking changes in our changelog (see "breaking" notation): https://github.com/maplibre/maplibre-gl-js/blob/main/CHANGELOG.md
Current breaking changes are more internal than actually API changes, we try and make sure we don't do a lot of breaking changes to our API, but some libraries are using some internal members and objects, which we have changed.
I hope this clarifies and answers your question.

**Reply by klokan** [reply]
I have set a Milestone 3.0.0 - I hope I did not miss a discussion where it would be confirmed that we don't want to use milestones on GitHub - as they were rarely used for previews releases... I expect just nobody did it, so I did it now.

**Reply by HarelM** [reply]
No problem, we usually are more kanban style, but since we are aiming for a breaking change version a milestone might be the right solution in this case.

**Comment by wipfli** [comment]
@roman-hatz-srf if you are also based in Switzerland, I am organizing a small meetup with people from Switzerland who are interested in MapLibre. It will take place on Monday, Jan 30th, 2022 at 18:30 and we will meet in Olten. Some people from swisstopo, SSB, and @bchapuis, @lseelenbinder, and @klokan might also be there.

If anyone would like to join, you can send me a message in Slack. You can get an invite at https://slack.openstreetmap.us and join the `#maplibre` channel, my Slack username is `@wipfli`...

**Reply by biodranik** [reply]
And from @OrganicMaps too )

**Comment by HarelM** [comment]
Due to the move of the style spec code to a different repo, the typings of it were moved as well.
This creates the extra effort of migration to 3.0 which isn't great.
The question is where should these typings be? Only in the style-spec package or also re-exported in maplibre?

**Reply by smellyshovel** [reply]
I vote for re-exporting just for the sake of convenience. Not crucial though, adding the `@maplibre/maplibre-gl-style-spec` dev. dep. and importing missing types from there works as well.

**Reply by HarelM** [reply]
Will be fixed once the #2453 PR is merged.

**Comment by HarelM** [comment]
Are there any last comments that someone would like to note before we release version 3.0?
cc @birkskyum
